api/instance/v1/instance_metadata_sdk.go

```diff
@@ -14,22 +14,41 @@ import (
 )
 
 var (
-	metadataURL           = "http://169.254.42.42"
 	metadataRetryBindPort = 200
 )
 
+const metadataAPIv4 = "http://169.254.42.42"
+const metadataAPIv6 = "http://[fd00:42::42]"
+
 // MetadataAPI metadata API
 type MetadataAPI struct {
+	MetadataURL *string
 }
 
 // NewMetadataAPI returns a MetadataAPI object from a Scaleway client.
 func NewMetadataAPI() *MetadataAPI {
 	return &MetadataAPI{}
 }
 
+func (meta *MetadataAPI) getMetadataUrl() string {
+	if meta.MetadataURL != nil {
+		return *meta.MetadataURL
+	}
+
+	for _, url := range []string{metadataAPIv4, metadataAPIv6} {
+		http.DefaultClient.Timeout = 3 * time.Second
+		resp, err := http.Get(url)
+		if err == nil && resp.StatusCode == 200 {
```

Comment:
What happens if err is not nil?

Reply:
The next one will be use. If no IP responde same behavior like today and default v4 is return.